DATM/AtmForce.F90

```diff
@@ -76,35 +86,48 @@ subroutine AtmForce(gcomp,exportState,externalClock,initmode,rc)
    
    call ESMF_TimeGet(nextTime,h_r8=hfwd,rc=rc)
   endif
-  write(msgString,'(3a,f12.3)')'using ',trim(filename),' at fwd clock hour ',real(hfwd,4)
+  write(msgString,'(3a,f12.3)')'using ',trim(filename),' at fwd clock hour ',hfwd
   call ESMF_LogWrite(trim(msgString), ESMF_LOGMSG_INFO)
 
-  ! read the Atm field data into the Fwd bundle
-  nfields = size(AtmBundleFields)
-  do ii = 1,nfields
-   if(AtmBundleFields(ii)%isPresent)then
-    varname = trim(AtmBundleFields(ii)%file_varname)
-
-    ! get the '_fwd' field
-    call ESMF_FieldBundleGet(AtmBundleFwd, &
-                             fieldName=trim(AtmBundleFields(ii)%field_name)//'_fwd', &
-                             field=field, rc=rc)
-    if (ChkErr(rc,__LINE__,u_FILE_u)) return
-
-    call ESMF_FieldRead(field, &
-                        fileName=trim(filename), &
-                        variableName = trim(varname), &
-                        rc=rc)
-    if (ChkErr(rc,__LINE__,u_FILE_u)) return
-
-    call ESMF_FieldGet(field,farrayPtr=AtmBundleFields(ii)%farrayPtr_fwd,rc=rc)
-    if (ChkErr(rc,__LINE__,u_FILE_u)) return
-
-     write(msgString,'(i6,2a18,f14.5)')ii,' inside AtmForce  ',trim(varname), &
-                             AtmBundleFields(ii)%farrayPtr_fwd(iprnt,jprnt)
-     call ESMF_LogWrite(trim(msgString), ESMF_LOGMSG_INFO)
-   endif !isPresent
-  enddo
+  fexists = .false.
+  if(mytask == 0)inquire(file=trim(filename), exist=fexists)
+  if(fexists) fpresent(1) = 1
+  call ESMF_VMBroadCast(vm, fpresent, count=1, rootPet=0, rc=rc)
+  if (ChkErr(rc,__LINE__,u_FILE_u)) return
+  if(fpresent(1) == 1)fexists = .true.
+
+  if(.not. fexists)then
+    if(mytask == 0)print *,'Cannot read file '//trim(filename)//': Aborting'
+    call ESMF_LogWrite('Cannot read file '//trim(filename)//': Aborting', ESMF_LOGMSG_INFO)
+    call ESMF_Finalize(endflag=ESMF_END_ABORT)
+  else
+    ! read the Atm field data into the Fwd bundle
+    nfields = size(AtmBundleFields)
+    do ii = 1,nfields
+     if(AtmBundleFields(ii)%isPresent)then
+      varname = trim(AtmBundleFields(ii)%file_varname)
+
+      ! get the '_fwd' field
+      call ESMF_FieldBundleGet(AtmBundleFwd, &
+                               fieldName=trim(AtmBundleFields(ii)%field_name)//'_fwd', &
+                               field=field, rc=rc)
+      if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+      call ESMF_FieldRead(field, &
+                          fileName=trim(filename), &
+                          variableName = trim(varname), &
+                          rc=rc)
+      if (ChkErr(rc,__LINE__,u_FILE_u)) return
```

Comment:
I'd suggest to add following for the error in this ESMF_FieldRead, if there is any issue when reading the file, the code should abort, currently it only return to the calling routine.

call ESMF_LogWrite('Missing input fields in datm_data_table', ESMF_LOGMSG_INFO)
call ESMF_Finalize(endflag=ESMF_END_ABORT)

Reply:
Yes, I will also add an abort if it tries to read the file but gets a return error.